crates/ruff_linter/src/linter.rs

```diff
@@ -214,6 +216,66 @@ pub fn check_path(
         ));
     }
 
+    // Raise violations for internal test rules
+    #[cfg(feature = "test-rules")]
+    {
+        if settings.rules.enabled(Rule::StableTestRule) {
+            diagnostics.push(Diagnostic::new(
+                crate::rules::ruff::rules::StableTestRule,
+                ruff_text_size::TextRange::default(),
+            ));
+        }
+
+        // Applying the fix must "resolve" the diagnostic or we will not converge
+        // So only raise the diagnostics on the "first" iteration
+        #[allow(clippy::used_underscore_binding)]
+        if _fix_iteration == 0 {
```

Comment:
Awkward, but this was really hard to do in the parent calling `check_path`.

Reply:
Maybe a comment explaining what `fix_iteration` is? What is the significance of its value? And what is the significance of `0`?

Reply:
I don't think this is right because we will exclude diagnostics that are not fixed... need to investigate